main.py

```diff
@@ -96,3 +98,56 @@ def plugin_unloaded():
         if ref_info:
             ref_view.settings().set('refinfo', ref_info.as_value())
     cli.service.exit()
+
+
+_UPDATE_NPM_MESSAGE = "Warning from TypeScript Plugin: We detect a global \
+npm typescript module and it is outdated. Consider upgrading this module by \
+running \"npm install -g typescipt\". TypeScript Plugin still works \
+regardless."
+
+def _check_typescript_version():
+    """
+    Notify user to upgrade npm typescript. Do this only once every time the
+    plugin is updated.
+    """
+    settings = sublime.load_settings('Preferences.sublime-settings')
+    cached_plugin_version = settings.get("typescript_plugin_tsc_version")
+
+    try:
+        plugin_tsc_version = _get_plugin_tsc_version()
+
+        if cached_plugin_version != plugin_tsc_version:
+            # The version number getting from the tsc command is different to
+            # the version number stored in the setting file. This means user
+            # has just updated the plugin.
+
+            npm_tsc_version = _get_npm_tsc_version()
+
+            if npm_tsc_version < plugin_tsc_version:
+                sublime.message_dialog(_UPDATE_NPM_MESSAGE)
+
+                # Update the version in setting file so we don't show this
+                # message twice.
+                settings.set("typescript_plugin_tsc_version", plugin_tsc_version)
+                sublime.save_settings("Preferences.sublime-settings")
+
+    except Exception as error:
```

Comment:
`subprocess.check_output()` throws exception if `node` or `tsc` is not in the PATH.